worlds/kdl3/Rules.py

```diff
@@ -0,0 +1,284 @@
+from worlds.generic.Rules import set_rule, add_rule
+from .Locations import location_table, level_consumables
+from .Names import LocationName, EnemyAbilities
+from .Items import copy_ability_table
+import typing
+from BaseClasses import MultiWorld
+
+if typing.TYPE_CHECKING:
+    from . import KDL3World
+    from BaseClasses import CollectionState
+
+def can_reach_level(state: "CollectionState", player: int, level: int, open_world: bool,
+                    ow_boss_req: int):
+    if level == 1:
+        return True
+    else:
+        if open_world:
+            return state.has(f"{LocationName.level_names_inverse[level - 1]} - Stage Completion", player, ow_boss_req)
+        else:
+            return state.has(f"{LocationName.level_names_inverse[level - 1]} 6 - Stage Completion", player)
```

Comment:
The way this works when `open_world` is disabled makes this game explode out into ~36 spheres. This can make generation needlessly slower as well as skew progression balancing. It looks like only 3 levels have logical requirements to complete them, so I don't think it's necessary to make every level logically dependent on an event like this, only those points where a logical requirement exists should be necessary.

Reply:
When I did rules on can_reach(Location), I was having generation failures because of it (I forget the specific reason, but it had something to do with sweep not accurately finding all locations that were reachable). I'll try to look into it again though.

Reply:
After investigation, changing the rule to be can_reach(Location) for `open_world` disabled leads to infrequent test failures seemingly caused by an issue addressed by #1764. As such, I'm going to avoid changing this until that PR is merged.

Reply:
Figured out a solution without needing 1764.